src/main/java/org/jenkinsci/plugins/oic/OicAlgorithmValidatorFIPS140.java

```diff
@@ -0,0 +1,129 @@
+package org.jenkinsci.plugins.oic;
+
+import com.nimbusds.jose.Algorithm;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.impl.AESCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ContentCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ECDHCryptoProvider;
+import com.nimbusds.jose.crypto.impl.PasswordBasedCryptoProvider;
+import com.nimbusds.jose.crypto.impl.RSACryptoProvider;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import jenkins.security.FIPS140;
+
+/**
+ * This class helps in validating algorithms for FIPS-140 compliance and filtering the non-compliant algorithms when in
+ * FIPS mode.
+ */
+public class OicAlgorithmValidatorFIPS140 {
+
+    private static final boolean isFIPSMode = FIPS140.useCompliantAlgorithms();
+    private static final Set<JWSAlgorithm> jwsSupportedAlgorithms = new LinkedHashSet<>();
+    private static final Set<JWEAlgorithm> jweSupportedAlgorithms = new LinkedHashSet<>();
+    private static final Set<EncryptionMethod> supportedEncryptionMethod = new LinkedHashSet<>();
+
+    // Below list of compliant algorithms will be used to block the FIPS non-compliant algorithms.
+    static {
+        // Init compliant Jws algorithms
+        jwsSupportedAlgorithms.addAll(MACSigner.SUPPORTED_ALGORITHMS);
+        jwsSupportedAlgorithms.addAll(RSASSASigner.SUPPORTED_ALGORITHMS);
+        jwsSupportedAlgorithms.addAll(ECDSASigner.SUPPORTED_ALGORITHMS);
+
+        // Init compliant Jwe algorithms
+        jweSupportedAlgorithms.addAll(AESCryptoProvider.SUPPORTED_ALGORITHMS);
+        jweSupportedAlgorithms.addAll(RSACryptoProvider.SUPPORTED_ALGORITHMS);
+        // RSA1_5 is deprecated and not a compliant algorithm.
+        jweSupportedAlgorithms.remove(JWEAlgorithm.RSA1_5);
+        jweSupportedAlgorithms.addAll(ECDHCryptoProvider.SUPPORTED_ALGORITHMS);
```

Comment:
seems to use [its own key derivation function](https://bitbucket.org/connect2id/nimbus-jose-jwt/src/94a2e4d11ce322966d30a551a2d482936b588c87/src/main/java/com/nimbusds/jose/crypto/impl/ECDHCryptoProvider.java?at=master#lines-104:139) ([ConcatKDF](https://bitbucket.org/connect2id/nimbus-jose-jwt/src/94a2e4d11ce322966d30a551a2d482936b588c87/src/main/java/com/nimbusds/jose/crypto/impl/ConcatKDF.java?at=master#ConcatKDF.java-48,70)) and so would not be compliant?